controllers/sync.go

```diff
@@ -15,65 +15,48 @@ import (
 	apiProxy "github.com/ytsaurus/ytsaurus-k8s-operator/pkg/apiproxy"
 )
 
-func canUpdateComponent(selector ytv1.UpdateSelector, component consts.ComponentType) bool {
-	switch selector {
-	case ytv1.UpdateSelectorNothing:
-		return false
-	case ytv1.UpdateSelectorMasterOnly:
-		return component == consts.MasterType
-	case ytv1.UpdateSelectorDataNodesOnly:
-		return component == consts.DataNodeType
-	case ytv1.UpdateSelectorTabletNodesOnly:
-		return component == consts.TabletNodeType
-	case ytv1.UpdateSelectorExecNodesOnly:
-		return component == consts.ExecNodeType
-	case ytv1.UpdateSelectorStatelessOnly:
-		switch component {
-		case consts.MasterType:
-			return false
-		case consts.DataNodeType:
-			return false
-		case consts.TabletNodeType:
-			return false
-		}
-		return true
-	case ytv1.UpdateSelectorEverything:
-		return true
-	default:
-		return false
+func canUpdateComponent(selectors []ytv1.ComponentUpdateSelector, component ytv1.Component) bool {
+	for _, selector := range selectors {
+		if selector.Class != consts.ComponentClassUnspecified {
+			switch selector.Class {
+			case consts.ComponentClassEverything:
+				return true
+			case consts.ComponentClassNothing:
+				return false
+			case consts.ComponentClassStateless:
+				if component.Type != consts.DataNodeType && component.Type != consts.TabletNodeType && component.Type != consts.MasterType {
+					return true
+				}
+			default:
+				return false
+			}
+		}
+		if selector.Component.Type == component.Type && (selector.Component.Name == "" || selector.Component.Name == component.Name) {
+			return true
+		}
 	}
+	return false
 }
 
 // Considers spec and decides if operator should proceed with update or block.
 // Block case is indicated with non-empty blockMsg.
 // If update is not blocked, updateMeta containing a chosen flow and the component names to update returned.
 func chooseUpdatingComponents(spec ytv1.YtsaurusSpec, needUpdate []components.Component, allComponents []components.Component) (components []ytv1.Component, blockMsg string) {
-	configuredSelector := spec.UpdateSelector
-	if configuredSelector == ytv1.UpdateSelectorUnspecified {
-		if spec.EnableFullUpdate {
-			configuredSelector = ytv1.UpdateSelectorEverything
-		} else {
-			configuredSelector = ytv1.UpdateSelectorStatelessOnly
-		}
-	}
-
+	configuredSelectors := getEffectiveSelectors(spec)
 	var canUpdate []ytv1.Component
 	var cannotUpdate []ytv1.Component
-	needFullUpdate := false
 
 	for _, comp := range needUpdate {
 		component := ytv1.Component{
-			Name: comp.GetName(),
+			Name: comp.GetShortName(),
 			Type: comp.GetType(),
 		}
```

Comment:
Woudln't it be easier if we use ytv1.Component for needUpdate and allComponents and store them that way in ComponentManager, so we convert them in one place?